zilliqa/src/crypto.rs

```diff
@@ -20,20 +22,44 @@ use sha3::{Digest, Keccak256};
 
 /// The signature type used internally in consensus, to e.g. sign block proposals.
 #[derive(Debug, Clone, Copy, PartialEq)]
-pub struct NodeSignature(bls_signatures::Signature);
+pub struct NodeSignature(Signature<Bls12381G2Impl>);
 
 impl NodeSignature {
     pub fn identity() -> NodeSignature {
-        NodeSignature(G2Affine::identity().into())
+        // Default to Basic signatures - it's the normal signature.
+        NodeSignature(Signature::<Bls12381G2Impl>::Basic(
+            blsful::inner_types::G2Projective::identity(),
+        ))
     }
 
     pub fn from_bytes(bytes: &[u8]) -> Result<NodeSignature> {
-        Ok(NodeSignature(bls_signatures::Signature::from_bytes(bytes)?))
+        // Default to Basic signatures - it's the normal signature.
+        Ok(NodeSignature(Signature::<Bls12381G2Impl>::Basic(
+            blsful::inner_types::G2Projective::from_compressed(bytes.try_into()?)
+                .expect("blst_p2_uncompress() error"),
+        )))
     }
 
     pub fn aggregate(signatures: &[NodeSignature]) -> Result<NodeSignature> {
-        let signatures: Vec<_> = signatures.iter().map(|s| s.0).collect();
-        Ok(NodeSignature(bls_signatures::aggregate(&signatures)?))
+        let signatures = signatures.iter().map(|s| s.0).collect_vec();
+
+        // IETF standards say N >= 1
+        // Handles single case where N == 1, as AggregateSignature::from_signatures() only handles N > 1.
```

Comment:
https://github.com/hyperledger-labs/agora-blsful/issues/10 - Feel free to reference here in case it gets fixed.